src/v/kafka/server/usage_aggregator.cc

```diff
@@ -347,7 +347,11 @@ void usage_aggregator<clock_type>::close_window() {
         } else {
             vlog(klog.info, "{}", err_str);
         }
-        cur.reset(now_ts);
+        vassert(
+          cur.begin % _usage_window_width_interval.count() == 0,
```

Comment:
are buckets restored from disk? what happens if we restore buckets from before enforcing alignment after upgrading to this code?

Reply:
In all cases where `.reset()` is called, an aligned timestamp is passed in, that's why I felt an assertion was the right choice here